### ISSUE TITLE
Expected catch() or return (promise/catch-or-return) fails on short syntax

### ISSUE THREAD

**Issue by dmig** [xjamundx/eslint-plugin-promise#73]
This code
```javascript
performRequest().then(() => console.debug('update successful'), err => console.error(err))
```
emits `Expected catch() or return (promise/catch-or-return)`

**Comment by xjamundx** [comment]
Are you using `{ allowThen: true }`?

**Comment by dmig** [comment]
No, just recommended config:
```javascript
...
  extends: [
    'standard',
    'plugin:vue/recommended',
    'plugin:promise/recommended',
  ],
  // required to lint *.vue files
  plugins: [
    'html', 'vue', 'promise'
  ],
...
```

**Comment by BjornMelgaard** [comment]
Fails even when I have `catch`

```js
import { createLogic } from 'redux-logic'
import * as currentUserActs from '~/app/actions/currentUser'
import * as appActs from '~/app/actions/app'
import { allTypes } from '~/app/utils'
import * as api from '~/chrome/shared/api'

const updateUser = createLogic({
  type: allTypes(currentUserActs.updateUser, appActs.storeCreated),
  process(_, dispatch, done) {
    api
      .get('/user/current')
      .then(resp => resp.data)
      .then(user =>
        dispatch(currentUserActs.updateUserSuccess(user)))
      .catch(err =>
        dispatch(currentUserActs.updateUserFailure(err)))
      .then(() => done())
  },
})

export default updateUser
```

```js
module.exports = {
  parser: 'babel-eslint',
  extends: ['airbnb', 'standard', 'plugin:promise/recommended'],
  settings: {
    'import/resolver': {
      // TODO: fix issue eslint-babel-plugin-root-import cant parse .babelrc.js
      // and convert .babelrc to .babelrc.js
      'babel-plugin-root-import': {},
    },
  },
  globals: {
    localStorage: true,
    chrome: true,
    URL: true,
    CHROME_EXTENSION_OAUTH_UID: true,
    CHROME_EXTENSION_OAUTH_SECRET: true,
    CHROME_EXTENSION_OAUTH_REDIRECT_URI: true,
  },
  rules: {
    // style
    'space-before-function-paren': ['error', 'never'],
    'comma-dangle': ['error', 'always-multiline'],
    quotes: ['error', 'single', { avoidEscape: true }],
    'no-multi-spaces': [
      'error',
      {
        exceptions: {
          Property: true,
          ImportDeclaration: true,
        },
      },
    ],
    'key-spacing': [
      'error',
      {
        mode: 'strict',
        align: 'value',
      },
    ],
    'arrow-parens': ['error', 'as-needed'],

    // other
    'no-console': 'off',
    'import/prefer-default-export': ['warn'],
    'global-require': ['off'],
    'react/jsx-filename-extension': ['error', { extensions: ['.js'] }],

    // TODO
    'react/prop-types': ['warn'],
    'react/forbid-prop-types': ['warn'],
  },
}
```

**Comment by xjamundx** [comment]
Fixed by #194
